### DISCUSSION TITLE
Paste option that automatically removes line breaks and double spaces

### DISCUSSION THREAD

**Discussion by Wolfowizz** [freeplane/freeplane#981] [Ideas]
First of all, thanks a lot for a great program! I am using it on a daily basis. My primary use of the software is to structure text (incl. collecting background information) before I write it up. So I copy a lot of information into my mind maps from PDFs, etc. It would be great to have an option to paste text into a new node and have line breaks and double spaces automatically removed. Would that be possible to implement? Thank you.

**Comment by euu2021** [comment]
I also copy a lot of information into FreePlane. Although it is possible to create a script that makes the adjustments in the text when pasting into FreePlane, I prefer using an AutoHotkey script, because then I can give different treatment depending on the source being copied. 

For example, only for PDFs I want to erase line breaks, so I use this script that is limited to text copied from PDF-XChangeEditor:

```
#IfWinActive ahk_exe PDFXEdit.exe
	$^c::
		Clipboard := ""
		Send ^c
		ClipWait 1
		if (ErrorLevel) {
			MsgBox 0x40010, Error, Cannot set clipboard contents.
			return
		}
		Clipboard := RegExReplace(Clipboard, "\R+", " ")
	return
#IfWinActive
```
([source](https://www.reddit.com/r/AutoHotkey/comments/rv0nb2/to_make_all_clipboard_text_into_one_line/))
